File: varuna-backend/app/schemas/ai.py
"""
AI Pydantic Schemas
"""

from typing import Optional, List
from pydantic import BaseModel
from app.schemas.map import Coordinates, MapRecommendation


class ChatMessageSchema(BaseModel):
    id: str
    sender: str
    text: str
    timestamp: str
    suggestedActions: Optional[List[str]] = None


class AiChatRequest(BaseModel):
    message: str
    conversationId: Optional[str] = None
    currentLocation: Optional[Coordinates] = None
    vesselId: Optional[str] = None


class AiChatResponse(BaseModel):
    conversationId: str
    message: ChatMessageSchema
    suggestedActions: Optional[List[str]] = None

File: Varuna-Frontend/src/data/api/ai.ts
/**
 * VARUNA AI Assistant API
 */

import { apiClient } from './client';
import { ChatMessage, VarunaInsight } from '../../domain/models/types';
import { Coordinates } from '../../domain/models/mapIntelligence';

export interface AiChatRequest {
  message: string;
  conversationId?: string;
  currentLocation?: Coordinates;
  vesselId?: string;
}

export interface AiChatResponse {
  conversationId: string;
  message: ChatMessage;
  groundedInsight?: VarunaInsight;
  suggestedActions?: string[];
}

export const aiApi = {
  async sendMessage(request: AiChatRequest): Promise<AiChatResponse> {
    return apiClient.post<AiChatResponse>('/ai/chat', request);
  },

  async getMarineInsight(location?: Coordinates): Promise<VarunaInsight> {
    return apiClient.post<VarunaInsight>('/ai/insight', {
      latitude: location?.latitude,
      longitude: location?.longitude,
    });
  },

  async getRecommendationExplanation(zoneId: string): Promise<{ explanation: string }> {
    return apiClient.get<{ explanation: string }>(`/ai/recommendation-explanation/${zoneId}`);
  },
};
